Jenkinsfile

```diff
@@ -37,65 +37,91 @@ void unittests(String version) {
-                             alwaysLinkToLastBuild: true,
-                             keepAll     : true,
-                             reportDir   : 'test-reports/coverage/',
-                             reportFiles : 'index.html',
-                             reportName  : "Coverage${version}",
-                             reportTitles: "Coverage${version}"])
-                    }
-                    step([$class: 'JUnitResultArchiver', testResults: 'test-reports/junit/*.xml'])
-                }
-            }
-        }
-    }
+				}
+				catch(ex) {
+					throw ex
+				}
+				finally {
+					if (fileExists('test-reports/coverage/index.html')) {
+						publishHTML([allowMissing: true, alwaysLinkToLastBuild: true, keepAll: true, reportDir: 'test-reports/coverage/', reportFiles: 'index.html', reportName: "Coverage${version}", reportTitles: "Coverage${version}"])
+					}
+					step([$class: 'JUnitResultArchiver', testResults: 'test-reports/junit/*.xml'])
+				}
+			}
+		}
+	}
 }
 
 try {
-    Map tasks = [ : ]
-
-    pythonVersion.each {
-        version ->
-            tasks[version] = { unittests(version) }
-    }
-
-    tasks['snyk'] = {
-        stage('snyk') {
-            Snyk snyk = new Snyk(this, bparams)
-            snyk.execute()
-        }
-    }
-
-    tasks['sonarqube'] = {
-        stage('sonarqube') {
-            SonarQube.execute(this, bparams)
-        }
-    }
-
-    tasks['nexusiq'] = {
-        stage('nexusiq') {
-            Nexusiq.execute(this, bparams)
-        }
-    }
-
-    parallel(tasks)
-
-    common.setResultIfNotSet(Constants.JSUCCESS)
-} 
-catch (ex) {
-    common.logException(ex)
-    common.setResultAbortedOrFailure()
-    throw ex
-} 
-finally {
-    common.setResultIfNotSet(Constants.JFAILURE)
-    buildsCommon.notifyPostBuild(bparams)
+	Map tasks = [: ]
+
+	pythonVersion.each {
+		version ->tasks[version] = {
+			unittests(version)
+		}
+	}
+
+	tasks['snyk'] = {
+		stage('snyk') {
+			Snyk snyk = new Snyk(this, bparams)
+			snyk.execute()
+		}
+	}
+
+	tasks['sonarqube'] = {
+		stage('sonarqube') {
+			SonarQube.execute(this, bparams)
+		}
+	}
+
+	tasks['nexusiq'] = {
+		stage('nexusiq') {
+			Nexusiq.execute(this, bparams)
+		}
+	}
+
+	tasks['runPylint'] = {
+		stage('runPylint') {
+			node(Constants.DOCKERNODE) {
+				withContainer(image: "python:3.6-buster", registry: '', inside: '-u root') {
+					try {
+						sh """
+						mkdir reports
+						pip install pylint
+                           			pylint --exit-zero --output-format=parseable --reports=n tenable > reports/pylint_tenable.log
+                           			pylint --exit-zero --output-format=parseable --reports=n tests > reports/pylint_tests.log
+                           			cat reports/pylint_tenable.log
+                           			cat reports/pylint_tests.log
+                        			"""
+					} catch(ex) {
+						throw ex
+					} finally {
+						if (fileExists('reports/pylint_tenable.log')) {
```

Comment:
This if makes no sense to me. If the linter dod not produce this file, obviously something went wrong and the build should fail. This will either happen via the recordIssues plugin not finding the file or an else condition that fails the build with an error.